vcf/io.go

```diff
@@ -0,0 +1,213 @@
+package vcf
+
+import (
+	"fmt"
+	"github.com/vertgenlab/gonomics/common"
+	"github.com/vertgenlab/gonomics/fileio"
+	"io"
+	"log"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+// Read parses a slice of VCF structs from an input filename. Does not store or return the header.
+func Read(filename string) ([]Vcf, Header) {
+	var answer []Vcf
+	file := fileio.EasyOpen(filename)
+	defer file.Close()
+	header := ReadHeader(file)
+	var curr Vcf
+	var done bool
+	for curr, done = NextVcf(file); !done; curr, done = NextVcf(file) {
+		answer = append(answer, curr)
+	}
+	return answer, header
+}
+
+// ReadToChan is a helper function of GoReadToChan.
+func ReadToChan(file *fileio.EasyReader, data chan<- Vcf, wg *sync.WaitGroup) {
+	for curr, done := NextVcf(file); !done; curr, done = NextVcf(file) {
+		data <- curr
+	}
+	file.Close()
+	wg.Done()
+}
+
+// GoReadToChan parses VCF structs from an input filename and returns a chan of VCF structs along with the header of the VCF file.
+func GoReadToChan(filename string) (<-chan Vcf, Header) {
+	file := fileio.EasyOpen(filename)
+	header := ReadHeader(file)
+	var wg sync.WaitGroup
+	data := make(chan Vcf)
+	wg.Add(1)
+	go ReadToChan(file, data, &wg)
+
+	go func() {
```

Comment:
I believe this go routine  can be eliminated by closing data at the comment above